### ISSUE TITLE
bug: 

### ISSUE THREAD

**Issue by runr02** [ionic-team/stencil#4231]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.21.0

### Current Behavior

Get build error when using top-level await. 

"Module format cjs does not support top-level await." 

### Expected Behavior

Solution should build successfully. 

### System Info

```shell
System: node 14.17.1

    Compiler: \@stencil\core\compiler\stencil.js
     Stencil: 2.21.0
  TypeScript: 4.9.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.16.1
```

### Steps to Reproduce

in ts file: 
`export class Hello{
  static async SayHello() {
    return Promise.resolve('Hello World');
  }
}

export const hello= await Profile.SayHello();
`

in another ts file import variable hello and use. Then build npm run build. You will receive the error. 

### Code Reproduction URL

see sample code above.

**Comment by rwaskiewicz** [comment]
Hey @runr02 👋 

Thanks for the issue! I don't believe this is an issue with Stencil specifically, as [CommonJS (cjs) does not support top-level await](https://github.com/nodejs/node/issues/21267). Can you describe your use case a little bit more for us?

**Comment by rwaskiewicz** [comment]
Thanks! I'm going to label this to get it into our backlog for the team to refine.

**Comment by cam-narzt** [comment]
To add to this, I was until recently building a stencil app which used top level await, and it built fine. So something changed and caused this to regress. What's even weirder is: I didn't change my app/package.json/package-lock.json i just reinstalled the modules with `npm i` due to the node_modules folder getting wiped and now it doesn't work :( From my git history I can say that it was definitely working April 3, and even reinstalling the node modules with `npm ci` instead of `npm i` doesn't help.

**Comment by HansClaasen** [comment]
> Here's git hub repo of starter project with replicated issue: https://github.com/runr02/toplevelawaitissue.git

I am able to get the component to render after applying this patch:

```patch
diff --git a/src/components/my-component/my-component.tsx b/src/components/my-component/my-component.tsx
index eb17b68..acb7f88 100644
--- a/src/components/my-component/my-component.tsx
+++ b/src/components/my-component/my-component.tsx
@@ -1,4 +1,4 @@
-import { Component, Prop } from '@stencil/core';
+import { Component, Prop, h } from '@stencil/core';
 // import { format } from '../../utils/utils';
 import { message } from '../../utils/toplevelawaitissue';

@@ -29,6 +29,6 @@ export class MyComponent {

   render() {
     // return <div>Hello, World! I'm {this.getText()}</div>;
-    return message;
+    return <div>{message}</div>;
   }
 }
```

Am I missing something here to reproduce this?

**Comment by runr02** [comment]
> > Here's git hub repo of starter project with replicated issue: https://github.com/runr02/toplevelawaitissue.git
> 
> @runr02 I am able to get the component to render after applying this patch:
> 
> ```diff
> diff --git a/src/components/my-component/my-component.tsx b/src/components/my-component/my-component.tsx
> index eb17b68..acb7f88 100644
> --- a/src/components/my-component/my-component.tsx
> +++ b/src/components/my-component/my-component.tsx
> @@ -1,4 +1,4 @@
> -import { Component, Prop } from '@stencil/core';
> +import { Component, Prop, h } from '@stencil/core';
>  // import { format } from '../../utils/utils';
>  import { message } from '../../utils/toplevelawaitissue';
> 
> @@ -29,6 +29,6 @@ export class MyComponent {
> 
>    render() {
>      // return <div>Hello, World! I'm {this.getText()}</div>;
> -    return message;
> +    return <div>{message}</div>;
>    }
>  }
> ```
> 
> Am I missing something here to reproduce this?

@HansClaasen it’s been a while and can take a closer look tomorrow but the issue presented itself when you build “npm run build”

**Comment by HansClaasen** [comment]
@runr02 you are right. I actually didn't run the build command and only called `npm start` and it turns out that the behavior in watch mode is different and the error is not thrown there.

After further investigation it turns out that the Stencil compiler always creates CJS artifacts by calling [`validateDist`](https://github.com/ionic-team/stencil/blob/85294e1eef7b440deb27dfcf5a978e29c8445bef/src/compiler/config/outputs/index.ts#L32). There doesn't seem to be any configuration that would prevent that.

I could see two options here:
1. Add a Stencil configuration option that disables the CJS properties for the `dist-lazy` and `dist-lazy-loader` output targets that are [added](https://github.com/ionic-team/stencil/blob/cb8d9a38afb3ede442282d9a7aa2b101534feedc/src/compiler/config/outputs/validate-dist.ts#L87-L112) in `validateDist`
2. catch the error thrown by [`generateCjs`](https://github.com/ionic-team/stencil/blob/5c85c332a7390b30fdba1d8598e618e5bf0f2b59/src/compiler/output-targets/dist-lazy/lazy-output.ts#L72) if it is a toplevel await error but keep throwing other errors

Any advice from the Stencil team would be appreciated. Thanks!
